python/pyspark/sql/udf.py

```diff
@@ -111,6 +111,10 @@ def returnType(self):
                 and not isinstance(self._returnType_placeholder, StructType):
             raise ValueError("Invalid returnType: returnType must be a StructType for "
                              "pandas_udf with function type GROUP_MAP")
+        elif self.evalType == PythonEvalType.SQL_PANDAS_GROUP_AGG_UDF \
+                and isinstance(self._returnType_placeholder, (StructType, ArrayType)):
```

Comment:
Hm .. i think we don't support `MapType` too?

Reply:
Good catch. Yes I added check for `MapType` too.